mesa/space.py

```diff
@@ -548,7 +550,428 @@ def exists_empty_cells(self) -> bool:
         return len(self.empties) > 0
 
 
-class SingleGrid(_Grid):
+def is_lambda_function(function):
+    """Check if a function is a lambda function."""
+    return (
+        inspect.isfunction(function)
+        and len(inspect.signature(function).parameters) == 1
+    )
+
+
+def is_numpy_ufunc(function):
```

Comment:
I think this function is unnecessary, it's not really shorter or easier to read

Reply:
https://github.com/projectmesa/mesa/pull/1898/commits/627eccb59777b43bd36e7f81c04a5b290f1a438e